hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/SubscriptionsOwnershipEndpoint.java

```diff
@@ -19,38 +22,44 @@
 @Path("subscriptions/owner")
 public class SubscriptionsOwnershipEndpoint {
 
-	private final OwnerSources ownerSources;
-	private final SubscriptionService subscriptionService;
-
-	@Autowired
-	public SubscriptionsOwnershipEndpoint(OwnerSources ownerSources,
-	                                      SubscriptionService subscriptionService) {
-		this.ownerSources = ownerSources;
-		this.subscriptionService = subscriptionService;
-	}
-
-	@GET
-	@Produces(APPLICATION_JSON)
-	@Path("/{ownerSourceName}/{ownerId}")
-	public List<Subscription> listForOwner(@PathParam("ownerSourceName") String ownerSourceName, @PathParam("ownerId") String id) {
-		OwnerId ownerId = resolveOwnerId(ownerSourceName, id);
-		return subscriptionService.listForOwnerId(ownerId);
-	}
-
-	@GET
-	@Produces(APPLICATION_JSON)
-	@Path("/{ownerSourceName}/{ownerId}/unhealthy")
-	public List<UnhealthySubscription> listUnhealthyForOwner(@PathParam("ownerSourceName") String ownerSourceName, @PathParam("ownerId") String id) {
-		OwnerId ownerId = resolveOwnerId(ownerSourceName, id);
-		return subscriptionService.listUnhealthyForOwner(ownerId);
-	}
-
-	private OwnerId resolveOwnerId(String ownerSourceName, String id) {
-		OwnerSource ownerSource = ownerSources.getByName(ownerSourceName)
-				.orElseThrow(() -> new OwnerSourceNotFound(ownerSourceName));
-		if (!ownerSource.exists(id)) {
-			throw new OwnerSource.OwnerNotFound(ownerSourceName, id);
-		}
-		return new OwnerId(ownerSource.name(), id);
-	}
+    private final OwnerSources ownerSources;
+    private final SubscriptionService subscriptionService;
+
+    @Autowired
+    public SubscriptionsOwnershipEndpoint(OwnerSources ownerSources,
+                                          SubscriptionService subscriptionService) {
+        this.ownerSources = ownerSources;
+        this.subscriptionService = subscriptionService;
+    }
+
+    @GET
+    @Produces(APPLICATION_JSON)
+    @Path("/{ownerSourceName}/{ownerId}")
+    public List<Subscription> listForOwner(@PathParam("ownerSourceName") String ownerSourceName, @PathParam("ownerId") String id) {
+        OwnerId ownerId = resolveOwnerId(ownerSourceName, id);
+        return subscriptionService.getForOwnerId(ownerId);
+    }
+
+    @GET
+    @Produces(APPLICATION_JSON)
+    @Path("/unhealthy")
```

Comment:
To call this endpoint we have to send http request with following path `/subscriptions/owner/unhealthy`. This suggests that this endpoint is directly related to subscriptions. What about topics? In the near future we would like to add logic which will inform us about unhealthy topics as well. Therefore, I suggest to extract this endpoint to a new class, for example `UnhealthyStatusEndpoint` with path `/unhealthy` and to call it we will have to send http request on a an address `http://{hermes-management-ip}:{port}/unhealthy`. This will allow us implement new features without changing API.